src/parser/asts/functions.c

```
@@ -138,3 +142,53 @@
+		if(t.type != STRING) {
+			return NULL;
+		}
+
+		char c;
+		while(c = *t.value++) {
+			buff[buffIndex] = c;
+			buffIndex++;
+
+			if(buffIndex >= buffSize) {
+				buffSize *= 1.5;
+				buff = realloc(buff, buffSize);
```

cppcheck (2.10-2)

[error] Common realloc mistake

(memleakOnRealloc)